### ISSUE TITLE
Consider VOLUME /var/lib/manticore

### ISSUE THREAD

**Issue by sanikolaev** [manticoresoftware/docker#82]
Think through it and think how it can help. E.g. can it alleviate the `EXTRA=1` thing?

**Comment by sanikolaev** [comment]
@djklim87 what advantages does it give? Should we updated the readme?

**Comment by djklim87** [comment]
@sanikolaev No. We just more stateful now. Our docs already show runs without volumes mappings

**Comment by skyscooby** [comment]
Why are there 2 x VOLUME statements in the Dockerfile now? /usr/local/lib/manticore and /var/lib/manticore ?

The documentation only says to mount the data volume on the second location, no mention of the first.. 
This causes docker to create a bunch of un-necessary anonymous volumes for the /usr/local/lib path .. (seems to be empty in my case). Do I need to create a new issue for this?

**Comment by sanikolaev** [comment]
> Why are there 2 x VOLUME statements in the Dockerfile now? /usr/local/lib/manticore and /var/lib/manticore ?

The other one was added within this issue in this commit https://github.com/manticoresoftware/docker/commit/8f8f2c879a309220cef6beb42465279735d925f1

@djklim87 since the change raises questions, I'm reopening the issue. Please explain the pros and cons of having the extra volume.

**Comment by skyscooby** [comment]
Thanks @sanikolaev this is more of a nuance than anything else.. it's not functionally critical.

If there is a good reason for /usr/local/lib/manticore to be volume I'd just ask for that to be added to the container documentation so we know why and what gets stored in it.. (I was questioning as my containers have not produced anything into these anonymous volumes through basic use). 

The challenge with adding VOLUME statements is that they can't easily be removed in downstream container builds if not desired.. (ones that use your official build as a FROM). It requires forking the entire project and removing them and rebuilding the container or using some hacky third party tools that re-write the docker metadata, which also breaks any signature stuff that might be present so better off forking.

When a Dockerfile declares a VOLUME but the user doesn't launch a container with a volume in the specified location then docker creates an 'anonymous' volume automatically.. I feel these 'anonymous' volumes are a bad way to manage stateful data that one might care about so I personally only use either explicitly named volumes or filesystem mappings (bind mounts) when containers have state. 

I think the 'anonymous' volumes are just a way to guarantee whatever volume driver dockerd is configured with is actually used instead of writing inside the container layer but I consider these volumes as ephemeral... generally declaring VOLUME is a heavy handed thing to do and most images don't do it as everything still works fine without them.. (you just need to explicitly declare your mapping in compose or on command line.

**Comment by djklim87** [comment]
Most databases use VOLUME directives, such as: [MySQL](https://hub.docker.com/layers/library/mysql/oraclelinux9/images/sha256-fc9f1cef19f31e4fbea0ce56af4d662108af4235b391100d1e8ad5aebbadb64e) [Clickhouse](https://hub.docker.com/layers/library/clickhouse/lts-focal/images/sha256-f7192ac051360ae08db75d330e3107b0f97c504225b4ac32f73f471c268fe20c) [Redis](https://hub.docker.com/layers/library/redis/8.0-M03-bookworm/images/sha256-abf63646812bfd09301c3b7072dcaac86231fbcdd649daea346244714d790af3) 


Why do we use VOLUME?
	•	It ensures data persistence by default, which is useful for beginners.
	•	It prevents accidental data loss when stopping or removing containers.
	•	Some users don’t manually specify volumes, so VOLUME acts as a safety net.

However, for advanced users like you, explicit volume management is always the better approach

**Comment by djklim87** [comment]
I removed one of the volumes that was no longer in use https://github.com/manticoresoftware/docker/commit/24cbbc892b783f07e62eb98e21b826de8fe35c5c

**Comment by skyscooby** [comment]
> I removed one of the volumes that was no longer in use [24cbbc8](https://github.com/manticoresoftware/docker/commit/24cbbc892b783f07e62eb98e21b826de8fe35c5c)

Thank you! That was the only issue I had was not knowing if I should or should not be mounting a second volume there! Cheers.
